demos/demo_imi_pq.cpp

```diff
@@ -121,7 +121,7 @@ int main ()
                 elapsed() - t0, nb);
 
         std::vector <float> database (nb * d);
-        std::vector <long> ids (nb);
+        std::vector <long long> ids (nb);
```

Comment:
This is not portable (though neither was the initial code). The right type here is `faiss::Index::idx_t`.